core/src/main/java/jenkins/install/InstallStateFilter.java

```diff
@@ -2,8 +2,8 @@
 
 import hudson.ExtensionList;
 import hudson.ExtensionPoint;
+import jakarta.inject.Provider;
 import java.util.List;
-import javax.inject.Provider;
```

Comment:
FYI this is a breaking API change because the public extension point uses `Provider` in its API.


` public abstract InstallState getNextInstallState(InstallState current, Provider<InstallState> proceed);`

Reply:
Sorry, I had not noticed that before, but you are right. Fortunately I could not find any affected open source plugins, and only a small handful of CloudBees plugins appear to be affected: `bluesteel`, `cloudbees-license-plugin`, and `cloudbees-assurance-plugin`. Since the migration is so easy (just change the imports) I am inclined to leave this as-is in both weeklies and LTS. Since we have to switch to Jakarta injection imports eventually to get to Guice 7, I believe that the benefits of releasing a version with full support for Jakarta injection imports (allowing us to eventually get to Guice 7) outweigh the small cost of adapting to this minor method signature change in a small handful of proprietary plugins.